lib/arclight/custom_document.rb

```diff
@@ -63,12 +64,22 @@ def arclight_field_definitions
         { name: 'repository', value: repository_as_configured(repository), index_as: :facetable }
       ]
     end
-    # rubocop: enable Metrics/MethodLength
+
+    def arclight_field_definitions_colls
+      [
+        { name: 'names_coll', value: names_coll, index_as: :symbol },
+        { name: 'access_subjects_coll', value: access_subjects_coll, index_as: :symbol }
+      ]
+    end
 
     def names
       [corpname, famname, name, persname].flatten.compact.uniq - repository
     end
 
+    def names_coll
+      names_array(%w[corpname famname name persname], parent: "archdesc[@level='collection']")
```

Comment:
We should remove the `[@level='collection']` part of the XPath - when I said "collection level" I meant at the top level of description, i.e. the `<archdesc/>`.